check rpm
not building docs (need oxygen)

8/19/2010 - created by Qing He <qing.he@intel.com>
1/24/2011 - Disable Fedora/Debian checking/settings by Mark Hatle <mark.hatle@windriver.com>
3/15/2011 - Updated to newer version of sat-solver by Mark Hatle <mark.hatle@windriver.com>

diff -ur git.orig/CMakeLists.txt git/CMakeLists.txt
--- git.orig/CMakeLists.txt	2011-03-15 17:14:22.000000000 -0500
+++ git/CMakeLists.txt	2011-03-15 17:20:09.648880770 -0500
@@ -35,22 +35,15 @@
 FIND_PACKAGE(Check REQUIRED)
 FIND_PACKAGE(ZLIB REQUIRED)
 
-IF ( FEDORA )
-MESSAGE(STATUS "Building for Fedora")
-ADD_DEFINITIONS( -DFEDORA )
-ENDIF ( FEDORA)
-
-IF ( DEBIAN )
-MESSAGE(STATUS "Building for Debian")
-ADD_DEFINITIONS( -DDEBIAN -DDEBIAN_SEMANTICS)
-ENDIF ( DEBIAN )
-
 IF ( MULTI_SEMANTICS )
 MESSAGE(STATUS "Enabling multi dist support")
 ADD_DEFINITIONS( -DMULTI_SEMANTICS)
 ENDIF ( MULTI_SEMANTICS )
 
-IF ( NOT DEBIAN )
+FIND_PACKAGE(PkgConfig REQUIRED)
+PKG_CHECK_MODULES(RPM REQUIRED rpm)
+INCLUDE_DIRECTORIES( ${RPM_INCLUDE_DIRS} )
+
 FIND_LIBRARY(RPMDB_LIBRARY NAMES rpmdb)
 IF ( NOT RPMDB_LIBRARY )
 FIND_LIBRARY(RPMDB_LIBRARY NAMES rpm)
@@ -59,13 +52,14 @@
 IF ( RPMIO_LIBRARY )
 SET( RPMDB_LIBRARY ${RPMIO_LIBRARY} ${RPMDB_LIBRARY} )
 ENDIF ( RPMIO_LIBRARY )
-IF ( FEDORA )
+FIND_LIBRARY(RPMMISC_LIBRARY NAMES rpmmisc)
+IF ( RPMMISC_LIBRARY )
+SET( RPMDB_LIBRARY ${RPMMISC_LIBRARY} ${RPMDB_LIBRARY} )
+ENDIF ( RPMMISC_LIBRARY )
 FIND_LIBRARY(DB_LIBRARY NAMES db)
 IF ( DB_LIBRARY )
 SET( RPMDB_LIBRARY ${DB_LIBRARY} ${RPMDB_LIBRARY} )
 ENDIF ( DB_LIBRARY )
-ENDIF ( FEDORA )
-ENDIF ( NOT DEBIAN )
 
 INCLUDE ( CheckFunctionExists )
 INCLUDE ( TestBigEndian )
@@ -164,7 +158,7 @@
 
 MESSAGE(STATUS "Looking modules in ${CMAKE_MODULE_PATH}")
 
-set ( CMAKE_C_FLAGS     "${CMAKE_C_FLAGS} -Werror -Wall" )
+set ( CMAKE_C_FLAGS     "${CMAKE_C_FLAGS} -Werror -Wall -g" )
 set ( CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS} -g -O3" )
 set ( CMAKE_C_FLAGS_DEBUG     "${CMAKE_C_FLAGS} -g3 -O0" )
 
@@ -173,7 +167,7 @@
 ADD_SUBDIRECTORY(tools)
 ADD_SUBDIRECTORY(tests)
 ADD_SUBDIRECTORY(examples)
-ADD_SUBDIRECTORY(doc)
+#ADD_SUBDIRECTORY(doc)
 
 MESSAGE(STATUS "version: ${VERSION}")
 
